### PR TITLE
Allow konflux images with oci index, be used in osbs

### PR DESCRIPTION
**Pull request by rcerven**
* STONEBLD-2896

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] Python type annotations added to new code
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request has a link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/fetch_sources.py [hunk 1]
```diff
@@ -683,9 +683,11 @@ def get_srpm_urls(self, sigkeys=None, insecure=False):
             self.log.debug('Resolving SRPM for RPM ID: %s', rpm_id)
 
             if rpm['external_repo_name'] != 'INTERNAL':
-                msg = ('RPM comes from an external repo (RPM ID: {}). '
-                       'External RPMs are currently not supported.').format(rpm_id)
-                raise RuntimeError(msg)
+                msg = ('RPM comes from an external repo (RPM ID: {}; NVR: {}). '
+                       'External RPMs are currently not supported, '
+                       'skipping').format(rpm_id, rpm['nvr'])
```
Comment (MartinBasti) [inline atomic_reactor/plugins/fetch_sources.py:688]: what are we skipping here?

File: atomic_reactor/config.py [hunk 2]
```diff
@@ -483,7 +483,7 @@ def hide_files(self):
     @property
     def skip_koji_check_for_base_image(self):
         return self._get_value(ReactorConfigKeys.SKIP_KOJI_CHECK_FOR_BASE_IMAGE_KEY,
-                               fallback=False)
+                               fallback=True)
```
Comment (MartinBasti) [inline atomic_reactor/config.py:486]: Shouldn't be this rather config change?

File: atomic_reactor/plugins/koji_parent.py [hunk 3]
```diff
@@ -23,7 +23,7 @@
 import time
 
 
-DEFAULT_POLL_TIMEOUT = 60 * 10  # 10 minutes
+DEFAULT_POLL_TIMEOUT = 60 * 3  # 3 minutes
```
Comment (MartinBasti) [inline atomic_reactor/plugins/koji_parent.py:26]: Why this change?

File: atomic_reactor/schemas/config.json [hunk 4]
```diff
@@ -497,7 +497,7 @@
         "type": "string"
     },
     "skip_koji_check_for_base_image": {
-        "description": "Allow base image without a related koji build (insecure) when there are no NVR labels set on the base image. If the NVR labels are set on the base image, the check is performed regardless",
+        "description": "Allow base image without a related koji build (insecure)",
         "type": "boolean",
         "default": false
```
Comment (MartinBasti) [inline atomic_reactor/schemas/config.json:502]: Default changed in config.py ?

File: atomic_reactor/plugins/fetch_sources.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/plugins/fetch_sources.py:688]: when I was testing some konflux image, they had some RPMs from external repo, so for that we can't get sources

File: atomic_reactor/plugins/koji_parent.py (hunk 3 above)
Comment (rcerven) [reply atomic_reactor/plugins/koji_parent.py:26]: it was case for build which didn't have brew build in koji

not sure why it was even 10m  I think 3m is just fine

File: atomic_reactor/config.py (hunk 2 above)
Comment (rcerven) [reply atomic_reactor/config.py:486]: it could be, but since RH is mainly using this, I'd say having default is fine as well

File: atomic_reactor/schemas/config.json (hunk 4 above)
Comment (rcerven) [reply atomic_reactor/schemas/config.json]: changed

File: atomic_reactor/plugins/koji_parent.py (hunk 3 above)
Comment (MartinBasti) [reply atomic_reactor/plugins/koji_parent.py:26]: unrelated change should be in separate commit

**Comment by MartinBasti** [comment]
mypy is failing

File: atomic_reactor/plugins/koji_parent.py (hunk 3 above)
Comment (rcerven) [reply atomic_reactor/plugins/koji_parent.py:26]: well it is sort of related

**Comment by rcerven** [comment]
> mypy is failing

well that is ci issue

**Comment by rcerven** [comment]
> mypy is failing

fixed

File: atomic_reactor/plugins/koji_parent.py (hunk 3 above)
Comment (MartinBasti) [reply atomic_reactor/plugins/koji_parent.py:26]: how?

Comment (rcerven) [reply atomic_reactor/plugins/koji_parent.py:26]: because konflux images don't have brew build, so this is just making it wait longer for non-existing koji build

Comment (MartinBasti) [reply atomic_reactor/plugins/koji_parent.py:26]: ah so if iamge is not in koji, then we will wait everytime 3 minutes?

Comment (rcerven) [reply atomic_reactor/plugins/koji_parent.py:26]: exactly, and 10m seemed quite a lot so I've lowered it

Comment (MartinBasti) [reply atomic_reactor/plugins/koji_parent.py:26]: I'm thinking if we even need this timeout, isn't  an leftover from autorebuilds feature which has been dropped?

Comment (rcerven) [reply atomic_reactor/plugins/koji_parent.py:26]: I'd leave it, 3m is fine

Comment (rcerven) [reply atomic_reactor/plugins/koji_parent.py:26]: it may happen that image used by build is still building, so we can give it 3m to finish

Comment (MartinBasti) [reply atomic_reactor/plugins/koji_parent.py:26]: okay, 3min sounds reasonable to finish build after image is pushed to registry in build process

**Review by MartinBasti — APPROVED** [review]

**Comment by MartinBasti** [comment]
Commit message could describe better what is changed and why

**Review by chmeliik — APPROVED** [review]
LGTM, assuming the handling for Konflux SBOMs will be a separate PR
